src/NServiceBus.Core/Pipeline/Incoming/InvokeHandlerTerminator.cs

```diff
@@ -49,6 +51,16 @@ await messageHandler
 
                 activity?.SetErrorStatus(ex);
 
+                //context.Extensions.SetOnRoot(ex.GetHashCode(), messageHandler.HandlerType);
+
+                if (messageHandler.Instance.GetType().GetInterface("IUseRecoverabilityConfiguration`1") is { } t)
+                {
+                    var configType = t.GetGenericArguments().Single();
+                    var config = (IRecoverabilityConfiguration)Activator.CreateInstance(configType);
+                    var recoverabilityConfiguration = config.OnError(ex, context.MessageBeingHandled, context.Headers, context.Extensions);
+                    context.Extensions.SetOnRoot(ex.GetHashCode().ToString(), recoverabilityConfiguration);
```

Comment:
Maybe we want to add some kind of a prefix here so we understand what this thing represents at all?

Reply:
yes the key definitely needs more clarity. The main intent here was to find a way to only apply the configuration in case the exception which was used to determine the configuration is the same exception that ends up in the recoverability pipeline.

Reply:
It's at least a little more revealing in terms of what it is. Ideally it would be the actual logical message but that wouldn't work

Reply:
> Ideally it would be the actual logical message but that wouldn't work

not sure I can follow, it passes the actual logical message (just in an untyped way)

Reply:
i meant the typed one ;) as in "PlaceOrderCommand"